user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -0,0 +1,424 @@
+    if (inputWithFailureSearch.isEmpty()) {
+      return InputValidation.passed();
+    }
+    return InputValidation.failed();
+  }
+
+  @Override
+  public boolean hasChanges() {
+    return !measurementFileItems.isEmpty();
+  }
+
+  private static class UploadItemsDisplay extends Div {
+
+    private final Div uploadSection;
+    private final Div uploadedItemsSection;
+    private final Div uploadedItemsDisplays;
+
+    public UploadItemsDisplay(Upload upload) {
+      var uploadSectionTitle = new Span("Upload the measurement data");
+      uploadSectionTitle.addClassName("section-title");
+
+      var saveYourFileInfo = new InfoBox().setInfoText(
+              "When uploading a tab-separated file, please save your Excel file as UTF-16 Unicode Text (*.txt) before uploading.")
+          .setClosable(false);
+
+      var restrictions = new Div();
+      restrictions.addClassName("restrictions");
+      restrictions.add(new Span("Supported file formats: .txt, .tsv, .xlsx"));
+      restrictions.add(
+          "Maximum file size: %s MB".formatted(MAX_FILE_SIZE_BYTES / Math.pow(1024, 2)));
+
+      this.uploadSection = new Div();
+      uploadSection.add(uploadSectionTitle, saveYourFileInfo, upload, restrictions);
+      uploadSection.addClassName("upload-section");
+
+      uploadedItemsSection = new Div();
+      uploadedItemsSection.addClassName("uploaded-items-section");
+
+      var uploadedItemsSectionTitle = new Span("Uploaded files");
+      uploadedItemsSectionTitle.addClassName("section-title");
+
+      uploadedItemsDisplays = new Div();
+      uploadedItemsDisplays.addClassName("uploaded-measurement-items");
+      uploadedItemsSection.add(uploadedItemsSectionTitle, uploadedItemsDisplays);
+      add(uploadSection, uploadedItemsSection);
+      addClassName("upload-items-display");
+    }
+
+    private void addFileToDisplay(MeasurementFileDisplay measurementFileDisplay) {
+      uploadedItemsDisplays.add(measurementFileDisplay);
+    }
+
+    private void removeFileFromDisplay(String fileName) {
+      MeasurementFileDisplay[] fileDisplays = fileDisplaysWithFileName(fileName);
+      uploadedItemsDisplays.remove(fileDisplays);
+    }
+
+    private MeasurementFileDisplay[] fileDisplaysWithFileName(String fileName) {
+      return uploadedItemsDisplays.getChildren()
+          .filter(MeasurementFileDisplay.class::isInstance)
+          .map(MeasurementFileDisplay.class::cast)
+          .filter(measurementFileDisplay -> measurementFileDisplay.measurementFileItem().fileName()
+              .equals(fileName))
+          .toArray(MeasurementFileDisplay[]::new);
+    }
+
+    public void hide() {
+      uploadedItemsDisplays.setVisible(false);
+    }
+
+    public void show() {
+      uploadedItemsDisplays.setVisible(true);
+    }
+  }
+
+  record MeasurementValidationReport(int validatedRows,
+                                     ValidationResult validationResult) {
+
+  }
+
+  public record MeasurementFileItem(String fileName,
```

Comment:
See my comment above, if there are multiple with the same filename, they must differ in the validation report. Otherwise there will be display problems as presumably vaadin uses the equals and hash code methods?

Reply:
I think this is out of scope for this work. However,  if you think this causes an actual issue for the users, please feel free and create an own issue.